### PR TITLE
Add `Files` API Module and expose the `upload` and `sharedPublicURL`

### PR DESCRIPTION
**Pull request by fullofcaffeine**

### CODE REVIEW

**Comment by fullofcaffeine** [comment]
@dempfi ping

File: packages/slack/src/api/files.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { APIModule } from './base'
+import { camel, snake } from './converters'
+import { Message, MessageOptions } from '../types'
+import * as messageFormat from '../helpers/formatters/message'
+
+export class Files extends APIModule {
+  /**
+   * https://api.slack.com/methods/files.upload
+   */
+  upload(options = {}) {
```
Comment (dempfi) [inline packages/slack/src/api/files.ts]: can you please add basic types for `options`

File: packages/slack/src/api/files.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return this.request('upload', snake(options)).then(camel)
+  }
+
+  /**
+   * https://api.slack.com/methods/files.sharedPublicURL
+   */
+  sharedPublicURL(options = {}) {
```
Comment (dempfi) [inline packages/slack/src/api/files.ts]: same here

**Comment by fullofcaffeine** [comment]
@dempfi Done.

**Comment by dempfi** [comment]
@fullofcaffeine this is awesome, thx for your contribution man

**Pull request merged**